### PR TITLE
Feature quickcheck

### PR DESCRIPTION
**Pull request by ililikakis**
**_Describe your changes here:_**

**_All these boxes must be checked by the reviewers before merging the pull request:_**

- [x] The author added a BSD statement to `doc/` (or already has one)
- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request indroduces code that is not covered by the github action (for example coupling with a new library):
  - [x] Should this use case be added to the github action?
  - [x] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [x] New source/header files are properly added to the Makefiles
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [x] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

**Comment by holke** [comment]
Thank you for your first t8code contribution :)

Very valuable input.
I will have a closer look at the code later this week.

File: test/t8_test_forest_commit.cxx [hunk 1]
```diff
@@ -127,7 +127,12 @@ t8_test_forest_commit (int cmesh_id)
   min_level = t8_forest_min_nonempty_level (cmesh, scheme);
   /* Use one level with empty processes */
   min_level = SC_MAX (min_level - 1, 0);
-  for (level = min_level; level < min_level + 3; level++) {
+#ifdef T8_ENABLE_LESS_TESTS
+  const int levelup = 2;
+#else
+  const int levelup = 3;
+#endif
+  for (level = min_level; level < min_level + levelup; level++) {
```
Comment (holke) [inline test/t8_test_forest_commit.cxx:135]: Was there a particular reason to change the levelup to 2?

I would rather leave it at 3. Since this level is not like the other as in "the original uniform level of the test mesh".
It specifies a refinement level for testing the adapt and balance routine.
Especially for the balance routine it is important to refine a few levels in order to catch possible boundary cases of balance (like neighboring elements of large level diff).

Hence, i would not touch this parameter for the less tests.

File: test/t8_test_ghost_and_owner.cxx [hunk 2]
```diff
@@ -125,7 +125,12 @@ t8_test_ghost_owner (int cmesh_id)
   min_level = SC_MAX (0, min_level - 1);
   t8_global_productionf
     ("Testing ghost exchange with start level %i\n", min_level);
-  for (level = min_level; level < min_level + 3; level++) {
+#ifdef T8_ENABLE_LESS_TESTS
+  const int levelup = 2;
+#else
+  const int levelup = 3;
+#endif
+  for (level = min_level; level < min_level + levelup; level++) {
```
Comment (holke) [inline test/t8_test_ghost_and_owner.cxx:133]: See my comment above. Would also rather keep this at 3.

File: test/t8_test_ghost_exchange.cxx [hunk 3]
```diff
@@ -186,7 +186,12 @@ t8_test_ghost_exchange (int cmesh_id)
   t8_global_productionf
     ("Testing ghost exchange start level %i. cmesh_id = %i\n", min_level,
      cmesh_id);
-  for (level = min_level; level < min_level + 3; level++) {
+#ifdef T8_ENABLE_LESS_TESTS
+  const int levelup = 2;
+#else
+  const int levelup = 3;
+#endif
+  for (level = min_level; level < min_level + levelup; level++) {
```
Comment (holke) [inline test/t8_test_ghost_exchange.cxx:194]: see comment above.

**Review by holke — CHANGES_REQUESTED** [review]
Great contribution.
This is a feature that is needed since we enlarged our test cases.

I have some minor corrections and comments but overall this looks very good.

i hope that #171 is resolved with the pyramids so that we can soon reduce the cmesh test cases.

**Comment by holke** [comment]
Comment: David said you would go through the above list and add anything missing from it in the next days. So i did not mention those points so far.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+build-aux/ltmain.sh
+build-aux/missing
+build-aux/test-driver
+config/libtool.m4
+config/ltoptions.m4
+config/ltsugar.m4
+config/ltversion.m4
+config/lt~obsolete.m4
+configure
+src/pre_config.h.in
```
[the remark this answers is not in the document]
Comment (ililikakis) [reply .gitignore]: Done.

File: src/t8_cmesh/t8_cmesh_testcases.c [hunk 5]
```diff
@@ -163,6 +173,9 @@ int
 t8_get_number_of_all_testcases ()
 {
   /* The number of all tests for all cmesh types is the sum of all individual testcase numbers. When disjoint bricks issue is fixed, then remove comment */
+#ifdef T8_ENABLE_LESS_TESTS
+  t8_debugf("change macros when issue #171 is solved");
+#endif
```
[the remark this answers is not in the document]
Comment (ililikakis) [reply src/t8_cmesh/t8_cmesh_testcases.c]: I added it to the TODO list in line 61.
commit 08e67714050a786a4195d700253492380a0c6ab4

File: test/t8_test_forest_commit.cxx (hunk 1 above)
Comment (ililikakis) [reply test/t8_test_forest_commit.cxx]: The reason was a time saving. I noticed that the 3rd levelup/forloop takes a lot of time. Especially the forest_commit takes much less time this way. To get a similar benefit, the cmesh test cases have to be reduced significantly.

t8_test_forest_commit takes about 80% of the time of the t8code test according to my rough measurements.

I understand your concerns however. I have not thought about that. 
The changes are reversed and once #17 is fixed, I will adjust the macros accordingly.

Comment (holke) [reply test/t8_test_forest_commit.cxx]: Thanks for clearing this up.
Its great that you did the timings of the tests and i see your reasoning.

I think your suggestion to wait for #171 and checking for the reduction in runtime is good.
Should this test case then still take too long we can reevaluate

**Comment by holke** [comment]
All looks good now.

As soon as its ready for review and the check boxes are addressed, we can merge.
@Davknapp i would leave the rest for you to decide then.

**Review by Davknapp — APPROVED** [review]
Thank you for your contribution!

**Review by holke — APPROVED** [review]
